### ISSUE TITLE
Cascade RPN with Cascade R-CNN

### ISSUE THREAD

**Issue by sehyungp92** [thangvubk/Cascade-RPN#15]
Firstly, many thanks for sharing this interest work.

I just had a very quick question on whether you were also able to see notable improvements when incorporating Cascade RPN with Cascade R-CNN, perhaps after increasing the IoU thresholds to reflect the higher quality of region proposals.

Kind regards,
Se

**Comment by thangvubk** [comment]
In the paper, we achieve 0.8 AP improvement. To be honest, we expect higher results when integrating Cascade RPN to Cascade R-CNN. Further study is needed on how to use high-quality proposal for Cascade R-CNN.

**Comment by sehyungp92** [comment]
Many thanks for your reply. Yes, it seems the use of high-quality proposal in Cascade R-CNN is not fully understood at the moment. Perhaps a combination or selection of adjusted IoU thresholds for the cascading stages, feature sharing between the box heads and Gaussian noise introduced to the region proposals.

On another note, I also found the part of the paper introducing Feature and Task Cascade very interesting, but when I clicked on the link it seems the github page was not longer available. Would this repo by any chance be made available once again?

Thanks.

**Comment by thangvubk** [comment]
Thank for your interest. We will release Feature and Task Cascade in the near future and notice here. :D

**Comment by sehyungp92** [comment]
Great, thanks!

**Comment by sehyungp92** [comment]
Apologies for the multiple questions, but very quickly, in the same paper, it mentions that the trained detector was finetuned with the precomputed proposals of Cascade RPN. Just wanted to quickly clarify:

- Does this involve training the data separately using a Cascade RPN then importing the proposals to the model? If so, how exactly do you import the computed proposals to a model?
- After importing the precomputed proposals, do the rest of configs for the model remain unchanged or do they also need to be adjusted (i.e. replace the RPN with a Cascade RPN)?

Many thanks.

**Comment by sehyungp92** [comment]
I promise this will be the last question from me! 

I'm currently undertaking a project where I believe the feature cascade component of the FTC may be particularly useful, and I was just wondering if you could kindly provide some details that may help with implementation and experimentation. In particular, I was wondering if the feature adaptation component was reminiscent of how the box head interacts with the mask head in MaskX R-CNN in Learn to Segment Every Thing (https://arxiv.org/pdf/1711.10370.pdf), or if it introduces a novel way for  the information flow between two branches.

Once again, many thanks in advance.
